api/datadoghq/common/datadogpodautoscaler_types.go

```diff
@@ -167,26 +190,97 @@ type DatadogPodAutoscalerContainerResourceObjective struct {
 	Container string `json:"container"`
 }
 
+// DatadogPodAutoscalerControllerObjective defines a controller-level objective
+// +kubebuilder:object:generate=true
+type DatadogPodAutoscalerControllerObjective struct {
+	Query TimeseriesFormulaRequest `json:"query"`
+	// Value is the value of the objective
+	Value DatadogPodAutoscalerObjectiveValue `json:"value"`
+}
+
+// TimeseriesFormulaRequest is a subset of the v2 timeseries query API (metrics only).
+// It mirrors the OpenAPI "TimeseriesFormulaRequestAttributes" fields relevant to autoscaling.
+// Reference: https://github.com/DataDog/datadog-api-spec/blob/94d1542b31ad0df1da915bae84686b13ba1a65ae/spec/v2/query.yaml#L124
+// +kubebuilder:object:generate=true
+type TimeseriesFormulaRequest struct {
+	// +kubebuilder:validation:Required
+	Attributes TimeseriesFormulaRequestAttributes `json:"attributes"`
+}
+
+// Spec mirrors the OpenAPI "TimeseriesFormulaRequestAttributes".
+// +kubebuilder:object:generate=true
+type TimeseriesFormulaRequestAttributes struct {
+	// Optional sampling interval in milliseconds.
+	// +optional
+	Interval *int64 `json:"interval,omitempty"`
+	// Formulas to compute (optional).
+	// +optional
+	Formulas []QueryFormula `json:"formulas,omitempty"`
+	// +kubebuilder:validation:MinItems=1
+	Queries []TimeseriesQuery `json:"queries"`
+}
+
+// +kubebuilder:object:generate=true
+type QueryFormula struct {
+	// +kubebuilder:validation:MinLength=1
+	Formula string `json:"formula"`
+}
+
+// TimeseriesQuery is a discriminated union. Only Metrics are supported for autoscaling.
+// +kubebuilder:validation:XValidation:rule="has(self.metrics)",message="metrics is required"
+// +kubebuilder:object:generate=true
+type TimeseriesQuery struct {
+	Metrics *MetricsTimeseriesQuery `json:"metrics,omitempty"`
+}
+
+// +kubebuilder:object:generate=true
+type MetricsTimeseriesQuery struct {
+	// Data source (Metrics API): "metrics" or "cloud_cost".
+	// +kubebuilder:validation:Enum=metrics;cloud_cost
+	DataSource MetricsDataSource `json:"dataSource"`
+	// Optional variable name ("a", "b", etc.) to reference in formulas.
+	// +optional
+	Name *string `json:"name,omitempty"`
+	// Classic Datadog metrics query, e.g. "avg:system.cpu.user{*} by {env}".
+	// +kubebuilder:validation:MinLength=1
+	Query string `json:"query"`
+}
+
+// +kubebuilder:validation:Enum=metrics;cloud_cost
+type MetricsDataSource string
+
+const (
+	MetricsDataSourceMetrics   MetricsDataSource = "metrics"
+	MetricsDataSourceCloudCost MetricsDataSource = "cloud_cost"
+)
+
 // DatadogPodAutoscalerObjectiveValueType specifies the type of objective value.
-// kubebuilder:validation:Enum:=Utilization
+// +kubebuilder:validation:Enum:=Utilization;AverageValue
```

Comment:
I'm not sure if `AverageValue` makes the most sense for external metrics. If we look at internal data, only 1.2% of WPAs are `average`, everything else is `absolute`.